src/path_string.rs

```diff
@@ -20,30 +20,31 @@ impl PathString for serde_json::Value {
   }
 }
 
-// TODO: improve performance
-impl PathString for EvaluationContext<'_> {
-  fn path_string(&self, path: &[String]) -> Option<Cow<'_, str>> {
+fn convert_value(value: &async_graphql::Value) -> Option<Cow<'_, str>> {
+  match value {
+    async_graphql::Value::String(s) => Some(Cow::Borrowed(s.as_str())),
+    async_graphql::Value::Number(n) => Some(Cow::Owned(n.to_string())),
+    async_graphql::Value::Boolean(b) => Some(Cow::Owned(b.to_string())),
+    async_graphql::Value::Object(map) => Some(json!(map).to_string().into()),
+    async_graphql::Value::List(list) => Some(json!(list).to_string().into()),
+    _ => None,
+  }
+}
+
+impl<'a, Ctx: GraphqlContext<'a>> PathString for EvaluationContext<'a, Ctx> {
+  fn path_string<T: AsRef<str>>(&self, path: &[T]) -> Option<Cow<'_, str>> {
     let ctx = self;
-    let mut result = None;
-    if let Some((head, tail)) = path.split_first() {
-      result = match head.as_str() {
-        "value" => ctx.path_value(tail).map(|v| v.to_owned()),
-        "args" => ctx.args()?.get_path(tail).cloned(),
-        "headers" => ctx.get_header_as_value(&tail[0]),
-        "vars" => Some(async_graphql::Value::String(
-          ctx.req_ctx.server.vars.clone()?.get(&tail[0]).cloned()?,
-        )),
+
+    path.split_first().and_then(|(head, tail)| {
+      assert!(tail.len() > 0);
+
+      match head.as_ref() {
+        "value" => convert_value(ctx.path_value(tail)?),
```

Comment:
I've noticed that the possible values in the code don't quite correspond to the values mentioned in the [docs](https://tailcall.run/docs/intro/context/).

Reply:
Yes, you are right. The documented Context exists only in the user-land. Internally we create an illusion of `Context` by shifting the responsibility to `EvaluationContext`. We could possibly create a `Context(&EvaluationContext)` which serves as a 1:1 map to the Context in the user-land.

Reply:
I'm not sure I get it.

I see that parts of the templates are replaced by `Mustache` struct like [here](https://github.com/tailcallhq/tailcall/blob/88a7b7c03419d531b7a37c37a8d50edd71d22ced/src/request_template.rs#L24) and there is only an `EvaluactionContext` that is used to paste the values in template